### PR TITLE
Fix code style violation of rule MEN002

### PR DESCRIPTION
**Pull request by v-pegao**
Workitem is [here](https://dev.azure.com/ceapex/Engineering/_workitems/edit/107202?src=WorkItemMention&src-action=artifact_link)

|Rule|Fix|
|---|---|
|MEN002: Lines too long<br/>(current line limit is 140 characters)|- If it is caused by too many nested brackets, try  turn local functions into member functions or use early return<br/>- If it is long parameter, try put parameters to a new line, if it is still too long, break each parameter into its own line<br/>- If parameter is a long expression, try extract a variable<br/>- For long member variable declarations with initialization, try put the default initializer to a new line<br/>- If the return value is a tuple ant the tuple definition is long, try put the return value definition in its own line|

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5023)

### CODE REVIEW

File: src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs [hunk 1]
```diff
@@ -40,19 +40,21 @@ public enum JavaScriptRuntimeAttributes
         /// <summary>
         ///     Using Eval or Function constructor will throw an exception.
         /// </summary>
-        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1650:ElementDocumentationMustBeSpelledCorrectly", Justification = "Eval is a valid function name.")]
+        [SuppressMessage(
+            "StyleCop.CSharp.DocumentationRules",
+            "SA1650:ElementDocumentationMustBeSpelledCorrectly",
```
Comment (v-pegao) [inline src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs]: I didn't find any SA1650 message after removed this, please help confirm if we still keep this.

Comment (yufeih) [reply src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs]: oh, don't touch the files under chakra folder, it is copied from elsewhere, there is a README.md explaining that

Comment (yufeih) [reply src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs]: There is a GlobalSuppressions.cs file that suppresses warnings in chakra folder

Comment (v-pegao) [reply src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs]: Got it, I will add MEN002 to there.

Comment (v-pegao) [reply src/docfx/lib/js/chakra/JavaScriptRuntimeAttributes.cs]: Reverted the changes for files under chakra folder

File: src/docfx/build/xref/ExternalXrefMapLoader.cs [hunk 2]
```diff
@@ -74,7 +74,8 @@ private static List<(string uid, long start, long end)> GetXrefSpecPositions(Rea
                 switch (reader.TokenType)
                 {
                     case JsonTokenType.PropertyName:
-                        if (reader.TextEquals(s_uidBytes) && reader.Read() && reader.TokenType == JsonTokenType.String && stack.TryPop(out var top))
+                        if (reader.TextEquals(s_uidBytes) && reader.Read()
```
Comment (yufeih) [inline src/docfx/build/xref/ExternalXrefMapLoader.cs]:
```suggestion
                        if (reader.TextEquals(s_uidBytes) && reader.Read() &&
                            reader.
```

File: src/docfx/lib/git/GitCommitProvider.cs [hunk 3]
```diff
@@ -9,12 +9,14 @@ namespace Microsoft.Docs.Build
 {
     internal sealed class GitCommitProvider : IDisposable
     {
-        private readonly ConcurrentDictionary<string, FileCommitProvider> _fileCommitProvidersByRepoPath = new ConcurrentDictionary<string, FileCommitProvider>();
+        private readonly ConcurrentDictionary<string, FileCommitProvider>
```
Comment (yufeih) [inline src/docfx/lib/git/GitCommitProvider.cs]: nit: vertically align the type signiture

```suggestion
        private readonly ConcurrentDictionary<string, FileCommitProvider> _fileCommitProvidersByRepoPath
                       = new  ConcurrentDictionary<string, FileCommitProvider>
```

File: src/docfx/lib/log/Telemetry.cs [hunk 4]
```diff
@@ -16,12 +16,21 @@ internal static class Telemetry
     {
         private static readonly TelemetryClient s_telemetryClient = new TelemetryClient();
 
-        private static readonly Metric s_operationTimeMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"time", "name", "os", "version", "repo", "branch"));
-        private static readonly Metric s_errorCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"error", "code", "level", "os", "version", "repo", "branch"));
-        private static readonly Metric s_cacheCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"cache", "name", "state", "os", "version", "repo", "branch"));
-        private static readonly Metric s_buildItemCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"count", "name", "os", "version", "repo", "branch"));
+        private static readonly Metric s_operationTimeMetric
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]:
```suggestion
        private static readonly Metric s_operationTimeMetric = s_telemetryClient.GetMetric(
            new MetricIdentifier(null, $"time", "name", "os", "version", "repo", "branch"));
```

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 5]
```diff
@@ -17,7 +17,10 @@ public static SourceInfo ToSourceInfo(this MarkdownObject obj, int? line = null,
         {
             // Line info in markdown object is zero based, turn it into one based.
             if (obj != null)
-                return new SourceInfo(file?.FilePath ?? (InclusionContext.File as Document)?.FilePath, obj.Line + 1, obj.Column + columnOffset + 1);
+            {
+                return new SourceInfo(
```
Comment (yufeih) [inline src/docfx/lib/markdown/MarkdigUtility.cs]: see if you can extract a local variable
```suggestion
                var file = file?.FilePath ?? (InclusionContext.File as Document)?.FilePath;
                return new SourceInfo(file, ...)
```

File: src/docfx/properties/GlobalSuppressions.cs [hunk 6]
```diff
@@ -4,27 +4,53 @@
...
-[assembly: SuppressMessage("Style", "SA1000", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1009", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1633", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1200", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1413", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1100", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1106", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
-[assembly: SuppressMessage("Style", "SA1116", Justification = "Others code", Scope = "namespaceanddescendants", Target = "ChakraHost.Hosting")]
+[assembly: SuppressMessage(
```
Comment (yufeih) [inline src/docfx/properties/GlobalSuppressions.cs]: can suppress SA1404 and remove `Justification=`

**Comment by yufeih** [comment]
@v-pegao any progress on this PR?

**Comment by herohua** [comment]
@v-pegao any progress on this PR?

**Comment by yufeih** [comment]
close stale pull requests

**Pull request closed without merging**